src/encord_active/app/common/components/metric_summary.py

```diff
@@ -28,13 +30,63 @@
 )
 from encord_active.lib.metrics.utils import (
     MetricData,
+    MetricSchema,
     MetricScope,
     load_available_metrics,
 )
 
 _COLUMNS = MetricWithDistanceSchema
 
 
+def render_2d_metric_plots(metrics_data_summary: MetricsSeverity):
+    with st.expander("2D metrics view", True):
+        metric_selection_col, scatter_plot_col = st.columns([2, 5])
+        metric_names = [metric_name for metric_name in metrics_data_summary.metrics.keys()]
+        x_metric_name = metric_selection_col.selectbox("x axis", metric_names, index=0)
```

Comment:
I would propose to use the `format_func` of the `selectbox` instead to directly get the MetricOutlierInfo class instead. Something like this (untested):

```suggestion
        x_outlier_info = metric_selection_col.selectbox("x axis", metrics_data_summary.metrics.values(), format_func=lambda m: m.name, index=0)
        x_metric_df = x_outlier_info.df[[MetricSchema.identifier, MetricSchema.score]]
```

This would make the code much cleaner.

Reply:
You can probably even get rid of the dictionary - and perhaps even sort the list on names to be very nice to the user :-)

Reply:
I did this as you suggested; however, it raised an internal streamlit error that I could not solve, spent nearly 2 hours on that, so I reverted it back. Here is the error output when I use format_func in selectbox:

<img width="1745" alt="image" src="https://user-images.githubusercontent.com/110481017/216361138-3ab861e3-223b-4590-8a89-86a35896f958.png">